SignalServiceKit/src/Messages/OWSDisappearingMessagesJob.m

```diff
@@ -225,27 +226,25 @@ - (void)becomeConsistentWithConfigurationForMessage:(TSMessage *)message
     [OWSDisappearingMessagesConfiguration fetchOrCreateDefaultWithThreadId:message.uniqueThreadId
                                                                transaction:transaction];
     
-    BOOL changed = NO;
     if (message.expiresInSeconds == 0) {
-        if (disappearingMessagesConfiguration.isEnabled) {
-            changed = YES;
-            DDLogWarn(@"%@ Received remote message which had no expiration set, disabling our expiration to become "
-                      @"consistent.",
-                      self.logTag);
-            disappearingMessagesConfiguration.enabled = NO;
-            [disappearingMessagesConfiguration saveWithTransaction:transaction];
-        }
-    } else if (message.expiresInSeconds != disappearingMessagesConfiguration.durationSeconds) {
-        changed = YES;
-        DDLogInfo(@"%@ Received remote message with different expiration set, updating our expiration to become "
-                  @"consistent.",
-                  self.logTag);
+        disappearingMessagesConfiguration.enabled = NO;
+    } else {
         disappearingMessagesConfiguration.enabled = YES;
         disappearingMessagesConfiguration.durationSeconds = message.expiresInSeconds;
-        [disappearingMessagesConfiguration saveWithTransaction:transaction];
     }
-    
-    if (!changed) {
+
+    if (!disappearingMessagesConfiguration.dictionaryValueDidChange) {
+        return;
+    }
+    DDLogInfo(@"%@ becoming consistent message configuration: %@",
+        self.logTag,
+        disappearingMessagesConfiguration.dictionaryValue);
+    [disappearingMessagesConfiguration saveWithTransaction:transaction];
+
+    TSThread *_Nullable thread = [message threadWithTransaction:transaction];
```

Comment:
If it's a thread created within this transaction, the sneaky transaction in message.thread won't see it, so the initial info message gets lost.